### PR TITLE
DC Local State Supplement - additional SNAP minimum allotment

### LINKED ISSUE #3610: DC Local State Supplement - additional SNAP minimum allotment
Mentioned [here](https://dhs.dc.gov/page/give-snap-raise-heres-what-expect):

>In the District of Columbia, the “Local State Supplement” (LSS) ensures that every SNAP household receives a minimum of $30 per month.

They state a $23 minimum allotment at https://dhs.dc.gov/service/snap-benefits but that seems incorrect and just copied from federal rules

Source: https://code.dccouncil.gov/us/dc/council/code/sections/4-261.04

@benmolin FYI - doesn't look like snapscreener has this (got $23/mo after entering an individual with $1,400/mo earnings)

### PR DESCRIPTION
**Pull request by JudyZ13**
Fix #3610

### CODE REVIEW

File: policyengine_us/variables/gov/usda/snap/dc_min_allotment.py
Comment (MaxGhenis) [inline policyengine_us/variables/gov/usda/snap/dc_min_allotment.py:1]: How about skipping this and putting the parameter in the main variable?

File: policyengine_us/variables/gov/usda/snap/snap_min_allotment.py [hunk 1]
```diff
@@ -24,4 +24,8 @@ def formula(spm_unit, period, parameters):
         # Minimum benefits only apply to households up to a certain size.
         size = spm_unit("spm_unit_size", period)
         eligible = size <= min_allotment.maximum_household_size
-        return eligible * min_allotment.rate * relevant_max_allotment
+        others_min = eligible * min_allotment.rate * relevant_max_allotment
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/snap_min_allotment.py:27]:
```suggestion
        min_allotment_outside_of_DC = eligible * min_allotment.rate * relevant_max_allotment
```

File: policyengine_us/variables/gov/usda/snap/snap_min_allotment.py [hunk 2]
```diff
@@ -24,4 +24,8 @@ def formula(spm_unit, period, parameters):
...
+
+        dc_min = spm_unit("dc_min_allotment", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/snap_min_allotment.py:29]:
```suggestion
        dc_min_alotment = spm_unit("dc_min_allotment", period)
```

File: policyengine_us/variables/gov/usda/snap/snap_min_allotment.py [hunk 3]
```diff
@@ -24,4 +24,8 @@ def formula(spm_unit, period, parameters):
...
+
+        return where(snap_region == 'DC', dc_min, others_min)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/snap_min_allotment.py:31]:
```suggestion
        state_code = spm_unit.household("state_code", period)
        in_dc = state_code == "DC"
        
        return where(in_dc, dc_min, others_min)
```

File: policyengine_us/variables/gov/usda/snap/dc_min_allotment.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    documentation = "DC SNAP minimum allotment"
+    label = "Minimum allotment for SNAP in DC"
+    unit = USD
+    defined_for = StateCode.DC
+
+    def formula(spm_unit, period, parameters):
+        # Parameters for the minimum benefit.
+        dc_min_allotment = parameters(period).gov.usda.snap.min_allotment.dc_amount
+        
+        return dc_min_allotment
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/dc_min_allotment.py:17]:
```suggestion
    adds = " parameters(period).gov.usda.snap.min_allotment.dc_amount"
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/usda/snap/dc_min_allotment.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+class dc_min_allotment(Variable):
+    value_type = float
+    entity = SPMUnit
+    definition_period = MONTH
+    documentation = "DC SNAP minimum allotment"
+    label = "Minimum allotment for SNAP in DC"
+    unit = USD
+    defined_for = StateCode.DC
+
+    adds = " parameters(period).gov.usda.snap.min_allotment.dc_amount"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/dc_min_allotment.py:13]:
```suggestion
    adds = "gov.usda.snap.min_allotment.dc_amount"
```

File: policyengine_us/variables/gov/usda/snap/snap_min_allotment.py [hunk 6]
```diff
@@ -24,4 +24,11 @@ def formula(spm_unit, period, parameters):
...
         eligible = size <= min_allotment.maximum_household_size
-        return eligible * min_allotment.rate * relevant_max_allotment
+        min_allotment_outside_of_DC = eligible * min_allotment.rate * relevant_max_allotment
+
+        dc_min_alotment = spm_unit("dc_min_allotment", period)
+
+        state_code = spm_unit.household("state_code", period)
+        in_dc = state_code == "DC"
+        
+        return where(in_dc, dc_min, others_min)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/snap_min_allotment.py:34]:
```suggestion
        return where(in_dc, dc_min_allotment, min_allotment_outside_of_DC)
```

File: policyengine_us/variables/gov/usda/snap/snap_min_allotment.py [hunk 7]
```diff
@@ -24,4 +24,11 @@ def formula(spm_unit, period, parameters):
         # Minimum benefits only apply to households up to a certain size.
         size = spm_unit("spm_unit_size", period)
         eligible = size <= min_allotment.maximum_household_size
-        return eligible * min_allotment.rate * relevant_max_allotment
+        min_allotment_outside_of_DC = eligible * min_allotment.rate * relevant_max_allotment
+
+        dc_min_alotment = spm_unit("dc_min_allotment", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/snap_min_allotment.py:29]:
```suggestion
        dc_min_allotment = spm_unit("dc_min_allotment", period)
```

File: changelog_entry.yaml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - DC Additional SNAP minimum allotment
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - DC Additional SNAP minimum allotment.
```

**Comment by MaxGhenis** [comment]
Please see #4569 also. We should probably fix that here and include an integration test to verify.

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**